docs/aws/audit/redshiftmonitoring/rules/emr_kerberos_enabled_remediation.mdx

```diff
@@ -1,6 +1,258 @@
+
+#### Example of Automating with AWS Lambda
+1. **Create a Lambda Function**:
+   - Go to the AWS Lambda console.
+   - Create a new Lambda function.
+   - Use the Python runtime.
+
+2. **Upload the Script**:
+   - Copy the script into the Lambda function editor.
+   - Adjust the script to handle multiple clusters if needed.
+
+3. **Set Up a Trigger**:
+   - Set up a CloudWatch Events rule to trigger the Lambda function periodically.
+
+By following these steps, you can ensure that Kerberos is enabled for your Amazon EMR clusters, thereby preventing misconfigurations related to security.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon Redshift dashboard by choosing "Redshift" under the "Database" category.
+3. In the navigation pane, choose "Clusters". The console displays a list of your current clusters.
+4. Select the cluster that you want to check. In the "Cluster Details" section, look for the "Cluster Properties" tab.
+5. Under the "Database configurations" section, check the "Kerberos" field. If it is not enabled, then the Amazon EMR cluster does not have Kerberos enabled.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can list all your Amazon EMR clusters using the following command:
+
+   ```
+   aws emr list-clusters --active
+   ```
+
+   This command will return a list of active EMR clusters in your AWS account.
+
+3. For each cluster, you can describe the cluster and check if Kerberos is enabled using the following command:
+
+   ```
+   aws emr describe-cluster --cluster-id <Cluster_ID>
+   ```
+
+   Replace `<Cluster_ID>` with the ID of your cluster. This command will return a JSON output with the details of the cluster.
+
+4. In the JSON output, look for the "SecurityConfiguration" field. If Kerberos is enabled, you should see a "KerberosAttributes" field with the details of the Kerberos configuration. If this field is not present, then Kerberos is not enabled on the cluster.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if Amazon EMR clusters have Kerberos enabled in Redshift using Python scripts, you can use the Boto3 library, which allows you to write software that makes use of services like Amazon S3, Amazon EC2, and others. Here are the steps:
+
+1. **Import the necessary libraries and establish a session with AWS:**
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>